### PR TITLE
Migrated all the moderation commands

### LINKED ISSUE #161: Migrating Moderation commands to slash command.
I will be migrating all the moderation commands to slash command. It might take some time like a week.

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
closes #161

### CODE REVIEW

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/AutoRolesCommand.java [hunk 1]
```diff
@@ -1,78 +1,74 @@
...
 	}
 
 	@Override
-	public void execute(final String[] args, final CommandEvent cmde, final MessageReceivedEvent e, final Message message, final BlackMember member, final BlackUser author, final BlackGuild guild, final TextChannel channel) {
-		final List<String> argsList = Arrays.asList(args);
-		if (argsList.contains("@everyone") || argsList.contains("@here")) {
-			cmde.error("invalidrole", "iseveryone");
-			return;
+	public void execute(@NotNull SlashCommandEvent cmde, @NotNull SlashCommandInteractionEvent e, BlackMember member,
+						BlackUser author, @NotNull BlackGuild guild, TextChannel channel) {
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/moderation/AutoRolesCommand.java:38]: Please make this a single line, thanks

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/AutoRolesCommand.java [hunk 2]
```diff
@@ -1,78 +1,74 @@
...
+			cmde.success("alreadyexisting", "thisalreadyexisting");
+			return;
+		} else
+			tempList.add(roleId);
+		guild.addAutoRole(roleId);
+		cmde.success("autorolecreated", "autorolecreatedinfo", new Placeholder("role", role.getAsMention()));
+	}
 
+	public void setRemoveCommand(@NotNull SlashCommandEvent cmde, @NotNull SlashCommandInteractionEvent e,
+								 @NotNull BlackGuild guild) {
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/moderation/AutoRolesCommand.java:61]: Same as above

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/AutoRolesCommand.java [hunk 3]
```diff
@@ -1,78 +1,74 @@
...
-				if (role != null) roleID = Long.parseLong(args[2]);
-			} catch (final NumberFormatException ignored) {
-			}
-			if (roleID == null) {
-				cmde.sendPleaseUse();
-				return;
-			}
-		}
+	public void setCreateCommand(@NotNull SlashCommandEvent cmde, @NotNull SlashCommandInteractionEvent e,
+								 @NotNull BlackGuild guild) {
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/moderation/AutoRolesCommand.java:47]: Same as above

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/ClearCommand.java [hunk 4]
```diff
@@ -1,47 +1,45 @@
...
-			if (amount == null || amount < 2 || amount > 100) {
-				cmde.send("toomanymessages" );
-				return;
-			}
 
 			try {
 				channel.getIterableHistory().cache(false).queue(msgs -> {
-					final OffsetDateTime firstValidDate = OffsetDateTime.now(ZoneOffset.UTC).minusWeeks(2).plusSeconds(1);
+					final OffsetDateTime firstValidDate = OffsetDateTime.now(ZoneOffset.UTC).minusWeeks(2)
+							.plusSeconds(1);
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/moderation/ClearCommand.java:42]: Please make a newline for either both minusWeeks and plusSeconds or dont use a newline

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/ClearCommand.java [hunk 5]
```diff
@@ -1,47 +1,45 @@
...
-			.setRequiredBotPermissions(Permission.MESSAGE_MANAGE));
+				.addOptions(new OptionData(OptionType.INTEGER, "amount", "Amount of messages to delete", true)
+						.setRequiredRange(2, 100))).setRequiredPermissions(Permission.MESSAGE_MANAGE).setEphemeral(true)
+								.setRequiredBotPermissions(Permission.MESSAGE_MANAGE));
 	}
 
 	@Override
-	public void execute(SlashCommandEvent cmde, SlashCommandInteractionEvent e, BlackMember member, BlackUser author, BlackGuild guild, TextChannel channel) {
+	public void execute(@NotNull SlashCommandEvent cmde, @NotNull SlashCommandInteractionEvent e, BlackMember member,
+			BlackUser author, BlackGuild guild, @NotNull TextChannel channel) {
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/moderation/ClearCommand.java:35]: Again, only one line for this please

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/ClearWarnCommand.java [hunk 6]
```diff
@@ -3,73 +3,73 @@
...
-				assert mentionedMember != null;
-				final List<Warn> warns = mentionedMember.getWarns();
-				for (final Warn warn : warns) {
-					if (warn.date() == warnId) {
-						mentionedMember.deleteWarn(warn);
-						cmde.success("entrydeleted", "warndeleted");
-						return;
-					}
+		} else {
+			var blackUser = BlackUser.from(e.getJDA().retrieveUserById(warnUser.getId()).submit().join());
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/moderation/ClearWarnCommand.java:62]: You can just use `warnUser` as the argument no?

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/ClearWarnCommand.java [hunk 7]
```diff
@@ -3,73 +3,73 @@
...
+			final List<Warn> userWarns = blackUser.getWarns();
+			for (final Warn warn : userWarns) {
+				if (warn.date() == warnId) {
+					blackUser.deleteWarn(warn);
+					cmde.success("warndeleted", "warndeleted");
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/moderation/ClearWarnCommand.java:67]: `cmde.send`

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/SuggestCommand.java [hunk 8]
```diff
@@ -3,43 +3,54 @@
...
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import org.jetbrains.annotations.NotNull;
 
-public class SuggestCommand extends TextCommand {
+public class SuggestCommand extends SlashCommand {
+	private static final String SUGGESTION = "suggestion";
 
 	public SuggestCommand() {
-		this.setCommand("suggest").setSyntax("<suggestion>").setRequiredArgumentCount(1);
+		super(builder(Commands.slash("suggest", "Used to of a suggestion").addOption(OptionType.STRING, SUGGESTION,
```
Comment (SIMULATAN) [inline src/main/java/com/github/black0nion/blackonionbot/commands/moderation/SuggestCommand.java:24]: Description: "Send a suggestion" or something similar

**Review by SIMULATAN — CHANGES_REQUESTED** [review]
Sorry for nitpicking :/

File: src/main/java/com/github/black0nion/blackonionbot/commands/moderation/AutoRolesCommand.java (hunk 1 above)
Comment (RealYusufIsmail) [reply src/main/java/com/github/black0nion/blackonionbot/commands/moderation/AutoRolesCommand.java:38]: Why? It makes uneat.

**Pull request merged**